.github/pull_request_template.md

```diff
@@ -0,0 +1,5 @@
+## Description of changes
+What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.
+
```

Comment:
I like how [embeddedartistry.com](https://embeddedartistry.com/blog/2017/8/4/a-github-pull-request-template-for-your-projects) has a checklist section on it.  I feel this would be a good thing for us.  We could use it for topics such as:

- [ ] Pull request is tested
- [ ] Code comments
- [ ] Update CHANGELOG.md
- [ ] I have updated the user documentation

Reply:
This is a good idea. For other ideas you can check out the Galaxy project checklist when merging tools https://github.com/galaxyproject/tools-iuc/pull/2132

For example:

FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/phac-nml/irida/blob/development/CONTRIBUTING.md) document.
* [ ] I have run a code formatter with the appropriate template for my IDE.
...

FOR REVIEWER:
* [ ] Includes tests or description of how to test out pull request.
* [ ] Code is well documented.
* [ ] If required, IRIDA documentation has been updated to describe this feature.
...

Reply:
Thanks guys.  I like the ideas.  How about keeping it slim for now, and increasing if needed?  I started with the following in 755c3af.  Some things like the code formatting might be good in the future.

```
* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [ ] Tests have been added for any new features.
* [ ] User documentation updated for UI or technical changes.
```

Reply:
Thanks, that looks good Tom.  I think it should be made clear who is filling out the checklist.  That is, is it the person submitting the pull request, or the reviewer? (you can add `for reviewer` or `for submitter` to make clear).

Reply:
@apetkau I like that, I think both would be good.

Reply:
Also, for tests. Is it too strong of a requirement to enforce tests to be made for any new features? For example, what about very difficult aspects to test (e.g., updating the tools/workflow for a pipeline, integration with external services, visualization systems, etc).  Should it instead be something like **Tests added (or description of how to test) any new features.**?

Reply:
Yeah agreed @apetkau .  I softened the test requirement a bit in 18d0ab38b3.  Some things just have to be manually tested.  Also added a blurb that the checklist is for the developer.

Reply:
Looks great. Thanks @tom114